src/arviz_plots/plots/essplot.py

```diff
@@ -0,0 +1,432 @@
+            _, div_aes, div_ignore = filter_aes(plot_collection, aes_map, "rug", sample_dims)
+            if "color" not in div_aes:
+                rug_kwargs.setdefault("color", "black")
+            if "marker" not in div_aes:
+                rug_kwargs.setdefault("marker", "|")
+            # WIP: if using a default linewidth once defined in backend/agnostic defaults
+            # if "width" not in div_aes:
+            #    # get default linewidth for backends
+            #    plot_bknd = import_module(f".backend.{backend}", package="arviz_plots")
+            #    default_linewidth = plot_bknd.get_default_aes("linewidth", 1, {})
+            #    rug_kwargs.setdefault("width", default_linewidth)
+            if "size" not in div_aes:
+                rug_kwargs.setdefault("size", 30)
+            div_reduce_dims = [dim for dim in distribution.dims if dim not in aux_dim_list]
+
+            # xname is used to pick subset of dataset in map() to be masked
+            xname = None
+            default_xname = sample_dims[0] if len(sample_dims) == 1 else "draw"
+            if (default_xname not in distribution.dims) or (
+                not np.issubdtype(distribution[default_xname].dtype, np.number)
+            ):
+                default_xname = None
+            xname = rug_kwargs.get("xname", default_xname)
+            rug_kwargs["xname"] = xname
+
+            draw_length = (
+                distribution.sizes[sample_dims[0]]
+                if len(sample_dims) == 1
+                else distribution.sizes["draw"]
+            )  # used to scale xvalues to between 0-1
+
+            plot_collection.map(
+                trace_rug,
+                "rug",
+                data=distribution,
+                ignore_aes=div_ignore,
+                # xname=xname,
+                y=distribution.min(div_reduce_dims),
+                mask=rug_mask,
+                scale=draw_length,
+                **rug_kwargs,
+            )  # note: after plot_ppc merge, the `trace_rug` function might change
+
+    # defining x_range (used for mean, sd, minimum ess plotting)
+    x_range = [0, 1]
+    x_range = xr.DataArray(x_range)
+
+    # plot mean and sd
+    if extra_methods is not False:
+        mean_kwargs = copy(plot_kwargs.get("mean", {}))
+        if mean_kwargs is not False:
+            mean_dims, mean_aes, mean_ignore = filter_aes(
+                plot_collection, aes_map, "mean", sample_dims
+            )
+            mean_ess = distribution.azstats.ess(
+                dims=mean_dims, method="mean", relative=relative, **stats_kwargs.get("mean", {})
+            )
+            print(f"\n mean_ess = {mean_ess}")
+
+            if "linestyle" not in mean_aes:
+                if backend == "matplotlib":
+                    mean_kwargs.setdefault("linestyle", "--")
+                elif backend == "bokeh":
+                    mean_kwargs.setdefault("linestyle", "dashed")
```

Comment:
you have to get the 2nd linestyle 

```
linestyle = plot_bknd.get_default_aes("linestyle", 2, {})[1]
```

or alternatively, the same way we have defined `|` must be a valid marker for all backends because it is part of the "common plotting interface" we could do the same for solid/dashed linestyles. For now I think I prefer the approach above though

Reply:
Linestyles -> mean, sd, min
Colors -> for diff models